diffkemp/simpll/DifferentialGlobalNumberState.h

```diff
@@ -28,6 +28,18 @@ class DifferentialGlobalNumberState : public GlobalNumberState {
     using ValueNumberMap = ValueMap<GlobalValue *, uint64_t>;
     ValueNumberMap GlobalNumbers;
 
+    // Mapping strings to numbers
+    StringMap<uint64_t> Strings;
+
+    // Mapping APInts to numbers
+    struct cmpConstants {
+        bool operator()(const APInt a, const APInt b) const {
```

Comment:
Use references (APInt &) which is more effective (the object does not have to be copied).